common/src/nostr/mod.rs

```diff
@@ -40,22 +48,58 @@ impl NostrClient {
             client,
             subscription_id: _subscription_id,
             notifications_receiver,
+            messages_cache: vec![],
         })
     }
 
     pub fn public_key(&self) -> PublicKey {
         self.keys.public_key()
     }
 
-    pub async fn receive_escrow_message(&mut self, timeout_secs: u64) -> anyhow::Result<String> {
+    pub async fn receive_escrow_message<T: DeserializeOwned>(
+        &mut self,
+        timeout_secs: u64,
+    ) -> anyhow::Result<T> {
+        let hit_idx_res = self
```

Comment:
Can you add a comment why the cache is necessary? May is not fully understandable on first look.

Reply:
Yes, good point. I'll add some explanation in the code. On master we still have the situation that when the buyer client runs before the seller, the seller fails receiving the registration from the coordinator, because it first get the token message already sent by the buyer. This was the initial motivation for this PR, making receiving messages type-safe fixes this error, but than you still have the problem, that if you discard the token message and just wait for the right one, you potentially wont get it later with our subscription. I don't unterstand completely why this error happens very reliably using production relays, but the nostr network is in general very fuzzy and makes only a few guaranties. We can't expect too much about message delivery rules. Messages will be posted several times and it is better also no to do assumptions about the order of the messages. I hope this throws some light on the topic.